charts/thanos/values.yaml

```diff
@@ -283,6 +283,87 @@ queryFrontend:
 
   affinity: {}
 
+receive:
+  # - Enable Receiver for Prometheus integration
+  enabled: false
+  serviceAccount:
+    create: true
+    labels: {}
+    annotations: {}
+    name: ""
+    automountToken: false
+
+  service:
+    annotations: {}
+
+  replicas: 1
+
+  updateStrategy: {}
+
+  podDisruptionBudget:
+    enabled: false
+    minAvailable:
+    maxUnavailable:
+
+  podLabels: {}
+
+  podAnnotations: {}
+
+  podSecurityContext:
+    fsGroup: 65534
+    runAsUser: 65534
+
+  securityContext: {}
+
+  priorityClassName: ""
+
+  terminationGracePeriodSeconds:
+
+  extraEnv: []
+
+  extraArgs: []
+  receiveAppLabels:
+    - name: "location"
+      value: "main"
+  tsdbRetention: "15d"
```

Comment:
Based on the other chart components strongly typed arguments live above `extraArgs`. Also are these flags required and are these appropriate values?

Reply:
This is mistake! I will remove these. as I was playing around

Reply:
The `receiveAppLabel` is used for tsdb db to separate databases
stated on documentation:
```
      --label=key="value" ...    External labels to announce. This flag will be
                                 removed in the future when handling multiple
                                 tsdb instances is added.
```
retention: https://thanos.io/tip/components/receive.md/#tenant-lifecycle-manage

with any of these missing, as stated. receiver do not start.

Reply:
I'm not sold on the labelling logic being deigned like this; have you looked at what they do for labels in [kube-thanos](https://github.com/thanos-io/kube-thanos)? I think `--label=receive="true"` & `--label=replica="$(NAME)"` would be the best defaults with end-users able to pass in any additional label arguments they wanted. Maybe with an extra value to disable default labels from being created to allow full control.

I also think `15d` for the default retention seems too high; I don't know what the cost of recreating a tenant is but I suspect it's likely low given that all data is pushed to object storage after `2h`. I'd be thinking something closer to `4h` or `6h` for the default.

Reply:
The label in thanos distinguishes between multiple receivers. 
And they require them to be set. (Datacenters etc)

receive="true" is a strange label to set on all your metrics. these are labels added into the stored metrics.

Reply:
It's been a while, let me re-read the docs for this point.

Reply:
some supporting reading: https://thanos.io/tip/operating/multi-tenancy.md/